x-pack/plugins/infra/server/lib/alerting/common/messages.ts

```diff
@@ -54,27 +58,36 @@ const comparatorToI18n = (comparator: Comparator, threshold: number[], currentVa
     case Comparator.LT:
       return ltText;
     case Comparator.GT_OR_EQ:
-    case Comparator.LT_OR_EQ:
-      if (threshold[0] === currentValue) return eqText;
-      else if (threshold[0] < currentValue) return ltText;
+    case Comparator.LT_OR_EQ: {
+      const currentValueAsNumber =
+        typeof currentValue === 'string' ? parseFloat(currentValue) : currentValue;
```

Comment:
This may be a nit, but do you think it would be better to parseFloat before calling this function? That way we can keep this function simple.